taichi/platform/opengl/opengl_kernel.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <taichi/common.h>
+#include <taichi/constants.h>
+#include <taichi/tlang_util.h>
+
+#include <string>
+#include <vector>
+
+#include "opengl_kernel_util.h"
+
+
+TLANG_NAMESPACE_BEGIN
+
+namespace opengl {
+
+struct SSBO
+{
+  void *const data;
```

Comment:
I would suggest using `std::vector<uint8_t> data(data_size)` so that memory allocation gets managed by RAII. Then you don't need to use the legacy C-style `malloc/free` anymore. You can get the starting of this memory buffer by `data.data()`.

Reply:
+1. I think with modern CPP, it should be very easy to avoid unnecessary `new/malloc`s...

Reply:
Good idea! Btw, the class `SSBO` will be completely removed by commits in later stages.